### ISSUE TITLE
Remove Auxiliary Variables in Conic Objective

### ISSUE THREAD

**Issue by ccoffrin** [lanl-ansi/PowerModels.jl#55]
Update, objective_min_fuel_cost{T <: AbstractConicPowerFormulation}(pm::GenericPowerModel{T}) in objective.jl.

The key is to define the min on a norm form.

**Comment by odow** [comment]
I'm not sure what this is asking for. I don't see any auxiliary variables, or where a norm form needs to be?

https://github.com/lanl-ansi/PowerModels.jl/blob/685fbb260fe1a773aef2a89dc205e0525b42a030/src/core/objective.jl#L18-L26

**Comment by ccoffrin** [comment]
I am open to dropping this after some thought.  It is worth double checking the findings of, https://github.com/lanl-ansi/PowerModels.jl/issues/353 with the latest version of JuMP.

The key code to look at is this (and maybe some similar variants),
https://github.com/lanl-ansi/PowerModels.jl/blob/master/src/core/objective.jl#L246-L279

When this was first developed (~2017) available conic solvers did not support quadratic objectives, so we lift the quadratic objective terms into new variables and a quadratic constraint to make the objective linear.  Around ~2020, this could be handled by JuMP bridges automatically.  In that case the `AbstractConicModels` objective function specialization in PowerModels could be removed in favor of JuMP bridges to do the lifting.  I tested this and there was a major performance degradation, not sure why (maybe bound reasoning on the lifted variables?).

The idea`min on a norm form` maybe another solution. I only have a vague recollection of what this is.  Need to discuss with a conic solver expert.

**Comment by odow** [comment]
JuMP now supports conic solvers with a quadratic objective. And if the solver doesn't support the objective, it automatically introduces the auxiliary variable, and if the solver doesn't support quadratic constraints, it automatically reformulates into the (Rotated)SecondOrderCone constraint.

**Comment by ccoffrin** [comment]
> And if the solver doesn't support the objective, it automatically introduces the auxiliary variable...

Yes, I tested this when I was exploring, https://github.com/lanl-ansi/PowerModels.jl/issues/353, but the performance was much worse than the current, by-hand implementation.  So I determined that relying on the automatic reformulation was not viable.  It has been a while so, worth checking this again with the latest version of JuMP/MOI.
